guides/HowToUseStreamDevice.md

```diff
@@ -0,0 +1,648 @@
+and all its contents
+and
+then to make the changes to `configure/RELEASE`
+and `configure/CONFIG_SITE` at mentioned above.
+
+Here are the commands I ran.
+
+
+```console
+norume> rm -rf configure
+norume> $EPICS_BASE/bin/$EPICS_HOST_ARCH/makeBaseApp.pl -t ioc HPE3631Atest
+norume> $EPICS_BASE/bin/$EPICS_HOST_ARCH/makeBaseApp.pl -t ioc -i HPE3631Atest
+The following target architectures are available in base:
+    darwin-x86
+    RTEMS-mvme2100
+    RTEMS-mvme3100
+    RTEMS-uC5282
+What architecture do you want to use? darwin-x86
+The following applications are available:
+    HPE3631Atest
+What application should the IOC(s) boot?
+The default uses the IOC's name, even if not listed above.
+Application name?
+norume>
+```
+
+I then added the ASYN
+and STREAM values to `configure/RELEASE`
+and made the change to the `CROSS_COMPILER_TARGET_ARCHS` line in `configure/CONFIG_SITE`.
+
+You must add the StreamDevice
+and ASYN support modules to the test application.
+Edit `HPE3631AtestApp/src/Makefile`
+and add the lines:
+
+
+```makefile
+HPE3631Atest_DBD += stream.dbd
+HPE3631Atest_DBD += asyn.dbd
+HPE3631Atest_DBD += drvAsynSerialPort.dbd
+```
+and
+```makefile
+HPE3631Atest_LIBS += stream asyn
+```
+
+To install the support module files
+and build the test application simply run `make`:
+
+```console
+norume> make
+```
+
+The IOC startup script produced by the `makeBaseApp.pl` script must be modified before the test application can be run.
+Change directories to `iocBoot/iocHPE3631Atest`
+and edit the `st.cmd` file.
+Here's the file that I use:
+
+
+```bash
+  1 #!../../bin/darwin-x86/HPE3631Atest
+  2
+  3 ###############################################################################
+  4 # Set up environment
+  5 < envPaths
+  6 epicsEnvSet "STREAM_PROTOCOL_PATH" "$(TOP)/db"
+  7
+
+  8 ###############################################################################
+  9 # Allow PV name prefixes and serial port name to be set from the environment
+ 10 epicsEnvSet "P" "$(P=hpE3631A)"
+ 11 epicsEnvSet "R" "$(R=Test)"
+ 12 epicsEnvSet "TTY" "$(TTY=/dev/tty.PL2303-000013FA)"
+ 13
+
+ 14 ###############################################################################
+ 15 ## Register all support components
+ 16 cd "${TOP}"
+ 17 dbLoadDatabase "dbd/HPE3631Atest.dbd"
+ 18 HPE3631Atest_registerRecordDeviceDriver pdbbase
+ 19
+
+ 20 ###############################################################################
+ 21 # Set up ASYN ports
+ 22 # drvAsynSerialPortConfigure port ipInfo priority noAutoconnect noProcessEos
+ 23 drvAsynSerialPortConfigure("L0","$(TTY)",0,0,0)
+ 24 asynSetOption("L0", -1, "baud", "9600")
+ 25 asynSetOption("L0", -1, "bits", "8")
+ 26 asynSetOption("L0", -1, "parity", "none")
+ 27 asynSetOption("L0", -1, "stop", "2")
+ 28 asynSetOption("L0", -1, "clocal", "Y")
+ 29 asynSetOption("L0", -1, "crtscts", "Y")
+ 30 asynOctetSetInputEos("L0", -1, "\\n")
+ 31 asynOctetSetOutputEos("L0", -1, "\\n")
```

Comment:
I have to check: are these escapes correct? We have had a lot of extra escape characters sneaking in...

Reply:
The colleagues here at codeathon agree that a single backslash should be sufficient.